Lesson7/organizer7/scheduler/views.py

```diff
@@ -0,0 +1,194 @@
+
+
+def alter_user(request):
+    global user
+
+    if request.method == "POST":
+        # Info from alter HTML-Form:
+        a_name = request.POST.get('username')
+        a_password = request.POST.get('password')
+        a_language = request.POST.get('language')
+        a_grade = get_grade(randint(1, 10))
+
+        if request.POST.get('action') == 'Next':
+            if check_user_in_db(a_name, a_password):
+                # User present in DB => update fields
+                # Avoiding New Instance Creation
+                db_user = User.objects.get(id=user.id)
+                db_user.name = a_name
+                db_user.password = a_password
+                db_user.language = a_language
+                db_user.grade = a_grade
+                # Updating info for forms
+                user.name = a_name
+                user.password = a_password
+                user.language = a_language
+                user.grade = a_grade
+
+                db_user.save()
+            else:
+                # Create a new user
+                user.name = a_name
+                user.password = a_password
+                user.language = a_language
+                user.grade = a_grade
+
+                user.save()
+
+            return redirect(show_profile)
+
+    # RENDER ALTER
+    return render(request, 'alter.html', {'user': user})
+
+
+def show_profile(request):
+    global user
+
+    if request.method == 'POST':
+        if request.POST.get('action') == 'Change User':
+            return redirect(alter_user)
+
+        if request.POST.get('action') == 'Delete User':
+            user = User.objects.filter(name=user.name, password=user.password).first()
+            user.delete()
+            return redirect(delete_profile)
+
+        # Create/Show NOTES page:
+        if request.POST.get('action') == 'NOTES':
+            return redirect(user_notes)
+
+    # RENDER PROFILE
+    return render(request, 'profile.html', {'user': user})
+
+
+def check_user_in_db(c_name, c_password):
+    global user
+
+    # FILTERED FIELDS FROM DB:
+    db_content = User.objects.filter(name=c_name, password=c_password).first()
+
+    if db_content:
+        # Assign DB user's fields to variables:
+        user.name = db_content.name
+        user.password = db_content.password
+        user.language = db_content.language
+        user.grade = db_content.grade
+        return True
+    return False
+
+
+def delete_profile(request):
+    if request.method == 'POST':
+        if request.POST.get('action') == 'Return to Login':
+            return redirect(login)
+
+    # RENDER DELETE
+    return render(request, 'delete.html')
+
+
+def get_grade(level):
+    if level < 4:
+        return 'Low'
+    elif level < 7:
+        return 'Medium'
+    else:
+        return 'High'
+
+
+def user_notes(request):
+    global user
+
+    # Check/Update user from DB:
+    user, created = User.objects.get_or_create(name=user.name)
```

Comment:
Creating a user while looking for their notes doesn't look like a good pattern since their no user notes yet, it would be better in that case to redirect to the logistration page and then propose user to start work with the notes